### PR TITLE
Fields add and delete, request from Advanced Search

### PR DESCRIPTION
**Pull request by alex-hornet**

### CODE REVIEW

**Comment by tarhunakau** [comment]
Could be merged

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 1]
```diff
@@ -3,101 +3,130 @@
 module.exports = function (search) {
 
     search.controller('searchCtrl', function ($scope, searchConfig, $uibModal, $stateParams, $state, promises, queryParams, searchStorage, searchService) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:5]: Are you sure that you need to many arguments?

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 2]
```diff
@@ -3,101 +3,130 @@
...
+                    reload: true
+                }
+            );
         };
-        
+
+
+        $scope.setObjectQuery = function (source, dest) {
+            var obj = searchStorage.objQuery.context[source];
+            delete(searchStorage.objQuery.context[source]);
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js]: Try with Factory Method

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 3]
```diff
@@ -131,9 +159,12 @@ module.exports = function (search) {
             params.offset = '0';
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js]: Offset is an integer value. Sometimes operations between Integers and String integers could be unpredictable

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 4]
```diff
@@ -105,25 +134,24 @@ module.exports = function (search) {
...
-            if (resultsTo > resultsCount){
+            if (resultsTo > resultsCount) {
                 resultsTo = resultsCount;
             }
             return resultsTo;
         }
 
         //If query params is empty set it to default
-        function setDefaultParams() {
+        $scope.setDefaultParams = function () {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:173]: Long and unreadable method

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 5]
```diff
@@ -141,55 +172,68 @@ module.exports = function (search) {
         $scope.sortParams = config.sortParams;
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:190]: _.extend?

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 6]
```diff
@@ -141,55 +172,68 @@ module.exports = function (search) {
...
-    
-    search.controller('searchDetailsCtrl', function($scope, searchConfig, searchStorage, $stateParams, $state, promises, searchService) {
-        
+
+    search.controller('searchDetailsCtrl', function ($scope, searchConfig, searchStorage, $stateParams, $state, promises, searchService) {
+
+        var params = {};
+
         // Convert details data for ng-repeat
         function convertDetails(details) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:246]: Array map?

arr.map(callback[, thisArg])

callback
    Функция, создающая элемент в новом массиве, принимает три аргумента:

```
currentValue
    Текущий элемент, создаваемый в массиве.
index
    Индекс текущего обрабатываемого элемента в массиве.
array
    Массив, по которому осуществляется проход.
```

thisArg
    Необязательный параметр. Значение, используемое в качестве this при вызове функции callback.

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js (hunk 1 above)
Comment (alex-hornet) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:5]: remove unnecessary

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js (hunk 2 above)
Comment (alex-hornet) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js]: Fixed

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 7]
```diff
@@ -3,101 +3,143 @@
...
+                    backUrl: location.hash
+                }, {
+                    inherit: true,
+                    reload: true
+                }
+            );
         };
+
+        $scope.getCurrentRequestContext = function () {
+            var obj = {};
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js]: ?!
var obj = {};
obj = {
  ...
}

what is that?

Comment (alex-hornet) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js]: Sorry, fixed.

File: angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js [hunk 8]
```diff
@@ -1,48 +1,87 @@
...
+            searchStorage.searchType = 'POST';
+
+            _.extend(
+                searchStorage.objQuery, 
+                $scope.buildRequest($scope.searchIn.value)
+            );
+
+            if ($scope.hasQuery()) {
+                $scope.showResults = false;
+                $uibModalInstance.close();
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js:33]: What is $uibModalInstance? Name of variable must be obvious

File: angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js [hunk 9]
```diff
@@ -1,48 +1,87 @@
...
+                    }
+                );
+            }
         };
-        
+
         // Function close modal window without request.
         $scope.cancel = function () {
-            $uibModalInstance.close($state.go('search.simpleQuery', searchStorage.params));
+            //$uibModalInstance.close($state.go('search.simpleQuery', searchStorage.params));
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js:46]: Do you really need an empty function that does nothing?

File: angular-spa-ui/js/search-app/search-module/search-config.js [hunk 10]
```diff
@@ -1,26 +1,49 @@
...
         ],
-        resultsPerPage: [
-            { title : '15 Results/page', value :'15' },
-            { title : '20 Results/page', value : '20' },
-            { title : '25 Results/page', value :'25' },
-            { title : '30 Results/page', value : '30' }
+    resultsPerPage: [
+        {
+            title: '15 Results/page',
+            value: '15'
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-config.js]: That is an integer, do not use strings

File: angular-spa-ui/js/search-app/search-module/search-config.js [hunk 11]
```diff
@@ -1,26 +1,49 @@
...
+        },
+        {
+            title: '20 Results/page',
+            value: '20'
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-config.js]: That is an integer, do not use strings

File: angular-spa-ui/js/search-app/search-module/search-config.js [hunk 12]
```diff
@@ -1,26 +1,49 @@
...
+        },
+        {
+            title: '25 Results/page',
+            value: '25'
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-config.js]: That is an integer, do not use strings

File: angular-spa-ui/js/search-app/search-module/search-config.js [hunk 13]
```diff
@@ -1,26 +1,49 @@
...
+        },
+        {
+            title: '30 Results/page',
+            value: '30'
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-config.js]: That is an integer, do not use strings

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 14]
```diff
@@ -3,101 +3,143 @@
...
+                conditions : $scope.query,
+                sortingOrder: "ASC",
+                sortingField: "title"
+            };
+            return obj;
+        }
+        
+        $scope.buildRequest = function (dest) {
+            var obj = {};
+            obj[dest] = $scope.getCurrentRequestContext();
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:39]: return {
  context: {
    dest: $scope.getCurrentRequestContext()
  }
};

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 15]
```diff
@@ -141,56 +185,69 @@ module.exports = function (search) {
         $scope.sortParams = config.sortParams;
         $scope.resultsPerPages = config.resultsPerPage;
         $scope.searchInList = config.searchIn;
-        
+
         if (searchService.isEmptyObject($scope.queryParams) || $scope.queryParams.query === undefined) {
-            $scope.queryParams = setDefaultParams();
+            $scope.queryParams = $scope.setDefaultParams();
             $scope.searchIn = $scope.searchInList[searchService.findValueId($scope.queryParams.searchIn, $scope.searchInList)];
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:197]: Very long line. Use index variable for searchInList.

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 16]
```diff
@@ -141,56 +185,69 @@ module.exports = function (search) {
         $scope.sortParams = config.sortParams;
         $scope.resultsPerPages = config.resultsPerPage;
         $scope.searchInList = config.searchIn;
-        
+
         if (searchService.isEmptyObject($scope.queryParams) || $scope.queryParams.query === undefined) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js]: Very long line.

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 17]
```diff
@@ -141,56 +185,69 @@ module.exports = function (search) {
...
-            $scope.queryParams = setDefaultParams();
+            $scope.queryParams = $scope.setDefaultParams();
             $scope.searchIn = $scope.searchInList[searchService.findValueId($scope.queryParams.searchIn, $scope.searchInList)];
             if (!searchService.isEmptyObject(searchStorage.data) && !searchService.isEmptyObject(searchStorage.params)) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:198]: Very long line.

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 18]
```diff
@@ -141,56 +185,69 @@ module.exports = function (search) {
...
+            if (searchStorage.searchType === 'GET') {
+                var queryUrl = queryParams
+                    .generateQueryParams(
+                        config.paths.simpleSearchPath,
+                        $scope.queryParams
+                    );
+            } else {
+                var queryUrl = config.paths.advancedSearchPath;
+            }
             $scope.searchIn = $scope.searchInList[searchService.findValueId($scope.queryParams.searchIn, $scope.searchInList)];
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:216]: Very long line.

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js (hunk 6 above)
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:246]: I still do no see changes

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js [hunk 19]
```diff
@@ -141,56 +185,69 @@ module.exports = function (search) {
...
-                inherit:true
+                inherit: true
             });
         }
     });
-    
-    search.controller('searchDetailsCtrl', function($scope, searchConfig, searchStorage, $stateParams, $state, promises, searchService) {
-        
+
+    search.controller('searchDetailsCtrl', function ($scope, searchConfig, searchStorage, $stateParams, $state, promises, searchService) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:241]: If function accepts more that 5 arguments - that is very bad solution

File: angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js (hunk 8 above)
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js:33]: What is $uibModalInstance? Name of variable must be obvious

File: angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js (hunk 9 above)
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js:46]: Do you really need an empty function that does nothing?

File: angular-spa-ui/js/search-app/search-module/advanced-search/index.js [hunk 20]
```diff
@@ -14,22 +14,27 @@ module.exports = function (angular) {
         $stateProvider
             .state('search.advanced', {
                 url: '/advanced',
+                params: {
+                    //searchIn : {squash: true , value: 'publication'}
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/advanced-search/index.js:18]: Remove commented code

Comment (alex-hornet) [inline angular-spa-ui/js/search-app/search-module/advanced-search/index.js:18]: Fixed

File: angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js (hunk 9 above)
Comment (alex-hornet) [inline angular-spa-ui/js/search-app/search-module/advanced-search/advanced-search-ctrl.js:46]: Fixed

File: angular-spa-ui/js/search-app/search-module/search-ctrl.js (hunk 19 above)
Comment (alex-hornet) [inline angular-spa-ui/js/search-app/search-module/search-ctrl.js:241]: This is Angular Way! ))

**Pull request merged**